lib/vrl/parser/src/template_string.rs

```diff
@@ -0,0 +1,76 @@
+use diagnostic::Span;
+use std::fmt;
+
+use crate::ast::{Expr, Ident, Literal::RawString, Node, Op, Opcode};
+
+#[derive(Clone, PartialEq, Eq, Ord, PartialOrd, Debug, Hash)]
+pub struct TemplateString(pub Vec<StringSegment>);
+
+impl TemplateString {
+    /// Rewrites the ast for the template string to be a series of string concatenations
+    pub fn rewrite(&self) -> Expr {
+        self.0
+            .iter()
+            .map(|node| -> Expr {
+                match node {
+                    StringSegment::Literal(s) => {
+                        Expr::Literal(Node::new(diagnostic::Span::default(), RawString(s.clone())))
```

Comment:
Is there a reason why we don't store the span in this variant, similar to the `Template` variant?

Reply:
We could store the span. I didn't because the only potential error reporting that requires spans occurs on the templated sections.

This should be easy enough to change to better future proof things.